### ISSUE TITLE
discrepancy between the expected(doc) and actual results with Pong+DQN

### ISSUE THREAD

**Issue by zhixiongzh** [opendilab/DI-engine#635]
- [x] I have marked all applicable categories:
    + [ ] exception-raising bug
    + [x] RL algorithm bug
    + [ ] system worker bug
    + [ ] system utils bug
    + [ ] code design/refactor
    + [ ] documentation request
    + [ ] new feature request
- [x] I have visited the [readme](https://github.com/opendilab/DI-engine/blob/github-dev/README.md) and [doc](https://opendilab.github.io/DI-engine/)
- [x] I have searched through the [issue tracker](https://github.com/opendilab/DI-engine/issues) and [pr tracker](https://github.com/opendilab/DI-engine/pulls)
- [x] I have mentioned version numbers, operating system and environment, where applicable:
  ```python
  import ding, torch, sys
  print(ding.__version__, torch.__version__, sys.version, sys.platform)
  ```
`v0.4.6 1.12.0 3.7.13 (default, Mar 29 2022, 02:18:16) 
[GCC 7.5.0] linux`

I used the code and configuration files provided in the repository to run the example "atari/pong+dqn" as described in the [example](https://di-engine-docs.readthedocs.io/zh_CN/latest/13_envs/atari_zh.html). However, the results I obtained were different from the expected results and took significantly longer to run.

here is the config file, I only change the exp name

```
pong_dqn_config = dict(
    exp_name='exp/pong/dqn_seed0',
    env=dict(
        collector_env_num=8,
        evaluator_env_num=8,
        n_evaluator_episode=8,
        stop_value=20,
        env_id='Pong-v4',
        #'ALE/Pong-v5' is available. But special setting is needed after gym make.
        frame_stack=4,
    ),
    policy=dict(
        cuda=True,
        priority=False,
        model=dict(
            obs_shape=[4, 84, 84],
            action_shape=6,
            encoder_hidden_size_list=[128, 128, 512],
        ),
        nstep=3,
        discount_factor=0.99,
        learn=dict(
            update_per_collect=10,
            batch_size=32,
            learning_rate=0.0001,
            target_update_freq=500,
        ),
        collect=dict(n_sample=96, ),
        eval=dict(evaluator=dict(eval_freq=4000, )),
        other=dict(
            eps=dict(
                type='exp',
                start=1.,
                end=0.05,
                decay=250000,
            ),
            replay_buffer=dict(replay_buffer_size=100000, ),
        ),
    ),
)
pong_dqn_config = EasyDict(pong_dqn_config)
main_config = pong_dqn_config
pong_dqn_create_config = dict(
    env=dict(
        type='atari',
        import_names=['dizoo.atari.envs.atari_env'],
    ),
    env_manager=dict(type='subprocess'),
    policy=dict(type='dqn'),
)
pong_dqn_create_config = EasyDict(pong_dqn_create_config)
create_config = pong_dqn_create_config
```
and my result
![pong_default dqn](https://img-blog.csdnimg.cn/ad716161ef1a4518ac9081970d0ad5c5.png)

Compared with the result showed in the doc, it is really confusing
![pong_dqn in doc](https://img-blog.csdnimg.cn/a05d9f63323a48bc961976c760f6be7a.png)

Could you please provide any guidance on how to troubleshoot the problem or suggest any known issues related to this example? Thank you.

**Comment by PaParaZz1** [comment]
We will conduct the reproducible experiment about this config in two days.

**Comment by PaParaZz1** [comment]
<img width="346" alt="Screen Shot 2023-04-10 at 7 11 11 PM" src="https://user-images.githubusercontent.com/33195032/230893006-8b0860d6-99c0-479d-ae31-8808b881f3d5.png">

As is shown above, I run the same config with 3 different seeds, the final results are similar to yours. I check the commit history and don't find the obvious config modifcation. In the next few days, I will check related code implementations and adjust some hyper-parameters. You can keep track of this issue.

**Comment by zhixiongzh** [comment]
> <img alt="Screen Shot 2023-04-10 at 7 11 11 PM" width="346" src="https://user-images.githubusercontent.com/33195032/230893006-8b0860d6-99c0-479d-ae31-8808b881f3d5.png">
> 
> As is shown above, I run the same config with 3 different seeds, the final results are similar to yours. I check the commit history and don't find the obvious config modifcation. In the next few days, I will check related code implementations and adjust some hyper-parameters. You can keep track of this issue.

Thanks for the effort!

It is great see your result. Actually your worst result with blue curse is also far better than mine as shown below. The orange curve is trained with the best ckpt from blue curve which was canceled due to accident. It still takes 14M steps. And for default PPO config(I also just change the exp name), it seems to be too slow, so I give up training longer.  
![reproducity](https://img-blog.csdnimg.cn/3b261aa546074ef391e5bf511c66ad5a.png)

**Comment by PaParaZz1** [comment]
I have checked the old log in server and find the curve shown in doc is the DQN+priority version. Then I validate this argument in 2 seeds and get the obvious improvement, which is shown in figure below. 
<img width="596" alt="Screen Shot 2023-04-11 at 11 58 49 PM" src="https://user-images.githubusercontent.com/33195032/231221366-1c1776df-b91d-4ac3-a94c-09d07dd2fb4e.png">

You can try to set `priority=True` in this config and rerun the experiment.

**Comment by PaParaZz1** [comment]
@zhixiongzh You should only set `priority=True` and keep `priority_IS_weight=False`. Because if we enable priority_IS_weight, we also need to adjust learning rate together.

**Comment by PaParaZz1** [comment]
I find a problem about the performance gap. In the old version, DI-engine uses `PongNoframeskip-v4` as env_id, but when we updated the version of gym==0.25.1, we modified this env_id to `Pong-v4`. This difference of these two env_ids is described in gym's doc:
<img width="853" alt="8MmZvTTshi" src="https://user-images.githubusercontent.com/33195032/236416602-12047028-4b6a-428a-b1ab-847dffd0f9be.png">

`Pong-v4` will bring some extra randomness to the env, so our original config will show some performance gap. I suggest you should modify the env_id and rerun the experiment. And I will also rerun it to check the result.
